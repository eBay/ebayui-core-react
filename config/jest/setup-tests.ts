import '@testing-library/jest-dom'

declare var global: typeof globalThis;

if (typeof window.URL.createObjectURL === 'undefined') {
  Object.defineProperty(window.URL, 'createObjectURL', { value: () => {} });
}
