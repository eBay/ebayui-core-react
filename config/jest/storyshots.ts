import { render } from "@testing-library/react";
import initStoryshotsAddon from "@storybook/addon-storyshots";
import type { StoryshotsOptions } from "@storybook/addon-storyshots/dist/ts3.9/api/StoryshotsOptions";

const reactTestingLibrarySerializer = {
    print: (val, serialize, indent) => serialize(val.asFragment()),
    test: (val) => val && val.hasOwnProperty("container"),
};

export function initStoryshots(config: StoryshotsOptions) {
    initStoryshotsAddon({
        ...config,
        renderer: render,
        snapshotSerializers: [reactTestingLibrarySerializer],
    });
}
