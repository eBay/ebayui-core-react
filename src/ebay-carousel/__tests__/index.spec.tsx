import React from 'react'
import requireContext from 'node-require-context'
import { initStoryshots } from '../../../config/jest/storyshots'
import { EbayCarouselItem, EbayCarousel } from "../index";
import { render, screen, cleanup } from "@testing-library/react";

// NOTE: need to mock scrollTo since JSDOM does not support it
jest.mock('../scroll-to-transition')

const items = Array(8).fill(0).map((_, i) => (
    <EbayCarouselItem key={i}>
        Item Text {i + 1}
    </EbayCarouselItem>
))

// Testing current slide or items per slides is not possible
// because we are using `getBoundingClientRect()` and when testing it returns 0 everytime
describe('<EbayCarousel />', () => {
    const a11yPreviousText = 'Previous Slide';
    const a11yNextText = 'Next Slide';

    beforeEach(() => {
        render(<EbayCarousel
        a11yPreviousText={a11yPreviousText}
        a11yNextText={a11yNextText}>
            {items}
        </EbayCarousel>)
    });

    afterEach(cleanup);

    it('shows the passed carousel items', () => {
        const visibleItems = screen.getAllByRole('listitem').filter(item => item.getAttribute("aria-hidden"))
        expect(visibleItems.length).toBe(8)

    })

    it('shows navigation sliding buttons', () => {
        const prevButton = getPrevSlideButton()
        const nextButton = getNextSlideButton()
        expect(prevButton).toBeInTheDocument()
        expect(prevButton).toHaveAttribute("class", "carousel__control carousel__control--prev")
        expect(nextButton).toBeInTheDocument()
        expect(nextButton).toHaveAttribute("class", "carousel__control carousel__control--next")
    })

    function getPrevSlideButton() {
        return screen.getByRole('button', { name: a11yPreviousText });
    }

    function getNextSlideButton() {
        return screen.getByRole('button', { name: a11yNextText });
    }
})

initStoryshots({
    config: ({ configure }) => {
        const req = requireContext('./', false, /\.stories\.tsx$/);
        return configure(req, module)
    }
});
