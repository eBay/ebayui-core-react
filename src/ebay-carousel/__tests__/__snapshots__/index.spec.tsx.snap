// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ebay-fullscreen-dialog Always Opened 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
  >
    <p
      aria-hidden="true"
    >
      Some outside content...
    </p>
    <div
      aria-hidden="false"
    >
      <div
        aria-hidden="false"
        aria-labelledby="dialog-title-abc123"
        arial-modal="true"
        class="fullscreen-dialog fullscreen-dialog--mask-fade-slow fullscreen-dialog--show-init"
        role="dialog"
      >
        <div
          aria-hidden="true"
          class="keyboard-trap-boundary"
          tabindex="0"
        />
        <div
          aria-hidden="false"
          class="fullscreen-dialog__window fullscreen-dialog__window--slide keyboard-trap--active"
        >
          <div
            aria-hidden="true"
            class="keyboard-trap-boundary"
            tabindex="0"
          />
          <div
            class="fullscreen-dialog__header"
          >
            <button
              aria-label="Close dialog"
              class="icon-btn fullscreen-dialog__close"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="icon icon--close"
                focusable="false"
                height="24px"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <use
                  xlink:href="#icon-close"
                />
              </svg>
            </button>
            <h2
              id="dialog-title-abc123"
            >
              Heading
            </h2>
          </div>
          <div
            class="fullscreen-dialog__main"
          >
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
            </p>
            <p>
              <a
                href="http://www.ebay.com"
              >
                www.ebay.com
              </a>
            </p>
          </div>
          <div
            class="fullscreen-dialog__footer"
          >
            Â©2021 eBay
          </div>
          <div
            aria-hidden="true"
            class="keyboard-trap-boundary"
            tabindex="0"
          />
        </div>
        <div
          aria-hidden="true"
          class="keyboard-trap-boundary"
          tabindex="0"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-fullscreen-dialog Default 1`] = `
<DocumentFragment>
  <div>
    <button
      class="btn btn--secondary"
    >
      Open Dialog
    </button>
    <p>
      Some outside content...
    </p>
    <div>
      <div
        aria-labelledby="dialog-title-abc123"
        arial-modal="true"
        class="fullscreen-dialog fullscreen-dialog--mask-fade-slow"
        hidden=""
        role="dialog"
      >
        <div
          class="fullscreen-dialog__window fullscreen-dialog__window fullscreen-dialog__window--slide"
        >
          <div
            class="fullscreen-dialog__header"
          >
            <button
              class="icon-btn fullscreen-dialog__close"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="icon icon--close"
                focusable="false"
                height="24px"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <use
                  xlink:href="#icon-close"
                />
              </svg>
            </button>
            <h2
              id="dialog-title-abc123"
            >
              Heading
            </h2>
          </div>
          <div
            class="fullscreen-dialog__main"
          >
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
            </p>
            <p>
              <a
                href="http://www.ebay.com"
              >
                www.ebay.com
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-fullscreen-dialog With Animation 1`] = `
<DocumentFragment>
  <div>
    <button
      class="btn btn--secondary"
    >
      Open Dialog
    </button>
    <p>
      Some outside content...
    </p>
    <div>
      <div
        aria-labelledby="dialog-title-abc123"
        arial-modal="true"
        class="fullscreen-dialog fullscreen-dialog--mask-fade-slow"
        hidden=""
        role="dialog"
      >
        <div
          class="fullscreen-dialog__window fullscreen-dialog__window fullscreen-dialog__window--slide"
        >
          <div
            class="fullscreen-dialog__header"
          >
            <button
              class="icon-btn fullscreen-dialog__close"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="icon icon--close"
                focusable="false"
                height="24px"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <use
                  xlink:href="#icon-close"
                />
              </svg>
            </button>
            <h2
              id="dialog-title-abc123"
            >
              Heading
            </h2>
          </div>
          <div
            class="fullscreen-dialog__main"
          >
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
            </p>
            <p>
              <a
                href="http://www.ebay.com"
              >
                www.ebay.com
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
